pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -179,6 +200,10 @@ func (n *NetworkPolicyController) updateNamespace(oldObj, curObj interface{}) {
 		affectedACNPsByOldLabels := n.filterPerNamespaceRuleACNPsByNSLabels(oldNamespace.Labels)
 		affectedACNPsByCurLabels := n.filterPerNamespaceRuleACNPsByNSLabels(curNamespace.Labels)
 		affectedACNPs := utilsets.SymmetricDifferenceString(affectedACNPsByOldLabels, affectedACNPsByCurLabels)
+		// Any ACNPs that has Namespace label rules that refers to the original or
```

Comment:
Why do we use `Union` here?
If NS label update is from `foo: bar` to `foo: bar, foo1: bar1`, the ACNP is 
```
appliedTo: 
    nsSel: foo=bar
ingress: 
    from:
       sameLabels: {foo}
```
Then it doesn't need to be re-processed right?

Reply:
For this specific case you can argue that it does not need to be re-processed. However, what if the Namespace label update is from `foo: bar` to `foo: bar1, foo1: bar`? At this stage I think it does not make sense to throw in additional logic to pre-filter, we can simply go through the ACNP processing to determine if there's any change in the processed internal NP

Reply:
I don't think it will be a complicated logic to introduce.  Maybe something like the below:
```
getMapDiffElements := func (map1, map2 map[string]string) map[string]string {
    diffElements := make(map[string]string)
    for k, v1 := range map1 {
        if v2, ok := map2[k]; !ok || v2 != v1 {
            diffElements[k] = v1
        }
    }
    for k, v2 := range map2 {
        if v1, ok := map1[k]; !ok || v1 != v2 {
            diffElements[k] = v2
        }
    }
    return diffElements
}
acnpsWithRulesMatchingNSLabelKeys := n.getNSLabelRuleACNPs(getMapDiffElements(oldNamespace.Labels, curNamespace.Labels))
```
It's still better to not enqueue them if we could filter them here?

Reply:
This is captured in `getACNPsWithRulesMatchingLabelKeysAcrossNSUpdate`.